### PR TITLE
With halt reboot

### PR DESCRIPTION
**Pull request by nledez**

### CODE REVIEW

File: client/app/assets/index.html [hunk 1]
```diff
@@ -26,20 +26,43 @@
...
-            <input type="button" id="button-refresh" class="btn btn-lg btn-default" value="Refresh information from Cozy">
-          </div>
-          <div id="div-reconfigure">
-            <input type="button" id="button-reconfigure" class="btn btn-lg btn-danger" value="Reconfigure my Cozy">
-          </div>
-          <div id="div-status-ko" class="alert alert-danger" role="alert"></div>
-          <div id="div-status-ok" class="alert alert-success" role="alert"></div>
-        </div>
+	<div id="content">
+        <div class="well">SELF HOSTING</div>
```
Comment (nledez) [inline client/app/assets/index.html]: Why change the case of this label?

File: client/app/assets/index.html [hunk 2]
```diff
@@ -26,20 +26,43 @@
...
+				<div id="div-reconfigure">
+					<input type="button" id="button-reconfigure" class="btn btn-lg btn-danger" value="Reconfigure my Cozy">
+				</div>
+			</div>
+		</div>
+
+<br /><br /><br />
+
+		<div class="panel panel-info">
+			<div class="panel-heading">HALT & REBOOT</div>
```
Comment (nledez) [inline client/app/assets/index.html]: I prefer the "Halt & reboot" form.

File: server/controllers/debian.js [hunk 3]
```diff
@@ -45,6 +60,71 @@ module.exports.update_fqdn = (req, res) => {
...
 };
+
+module.exports.host_halt = (req, res) => {
+    // halt the host
+
+    var exec = require('child_process').exec,
+        child;
+
+    const config_filename = '/etc/cozy/self-hosting.json';
+    var halt_script = '/usr/local/sbin/debian-halt.sh';
```
Comment (nledez) [inline server/controllers/debian.js]: Look the `var reconfigure_command = 'sudo ' + reconfigure_script` form to allow different path.

File: to-install/debian-halt.sh [hunk 4]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+if [ "$1" = "reboot" ]; then
```
Comment (nledez) [inline to-install/debian-halt.sh]: May be create two different scripts.

File: client/app/assets/index.html [hunk 5]
```diff
@@ -26,20 +26,43 @@
...
+			<div class="panel-body">
+				<div class="row">
+					<div class="col-md-6"><a id="button-halt" class="btn btn-lg btn-danger" title="">Halt the host</a></div>
```
Comment (nledez) [inline client/app/assets/index.html]: Can you add the ability to hide the button if parameter missing in config file?
Halt may be a bad idea on VPS.

File: client/app/assets/index.html (hunk 1 above)
Comment (SomeAverageDev) [inline client/app/assets/index.html]: I changed it back, I prefer upper case, but it's personnal

File: client/app/assets/index.html (hunk 2 above)
Comment (SomeAverageDev) [inline client/app/assets/index.html]: done

File: server/controllers/debian.js (hunk 3 above)
Comment (SomeAverageDev) [inline server/controllers/debian.js]: sorry I don't understand, I used the same code as the fqdn function...
I changed all the paths to "/usr/local/sbin/", and it is configurable through the file /etc/cozy/self-hosting.json

File: to-install/debian-halt.sh (hunk 4 above)
Comment (SomeAverageDev) [inline to-install/debian-halt.sh]: done

File: client/app/assets/index.html (hunk 5 above)
Comment (SomeAverageDev) [inline client/app/assets/index.html]: done

**Comment by frankrousseau** [comment]
@nledez can we merge that one?

**Pull request merged**